rotkehlchen/chain/ethereum/modules/eigenlayer/decoder.py

```diff
@@ -0,0 +1,108 @@
+        strategy = hex_or_bytes_to_address(context.tx_log.data[64:96])
+
+        for event in context.decoded_events:
+            if (
+                event.event_type == HistoryEventType.SPEND and
+                event.asset == token_identifier and
+                event.location_label == depositor
+            ):
+                event.event_type = HistoryEventType.STAKING
+                event.event_subtype = HistoryEventSubType.DEPOSIT_ASSET
+                asset = event.asset.resolve_to_crypto_asset()
+                event.notes = f'Deposit {event.balance.amount} {asset.symbol} in EigenLayer'
+                event.extra_data = {'strategy': strategy}
+                event.product = EvmProduct.STAKING
+                event.counterparty = CPT_EIGENLAYER
+
+        return DEFAULT_DECODING_OUTPUT
+
+    def _decode_withdrawal(self, context: DecoderContext) -> DecodingOutput:
+        """
+        Decode withdrawal from eigenlayer. The transaction only contains a transfer event
+        and the unstake event but the unstake event doesn't have information about the asset
+        or the amount unstaked.
+        """
+        depositor = hex_or_bytes_to_address(context.tx_log.topics[1])
+        withdrawer = hex_or_bytes_to_address(context.tx_log.topics[2])
+        depositor_is_tracked = self.base.is_tracked(depositor)
+        withdrawer_is_tracked = self.base.is_tracked(withdrawer)
+
+        if depositor_is_tracked is True and withdrawer_is_tracked is True:
+            event_type, event_subtype = HistoryEventType.STAKING, HistoryEventSubType.REMOVE_ASSET
+            notes = 'Unstake {amount} {symbol} at EigenLayer'
+        elif depositor_is_tracked is False and withdrawer_is_tracked is True:  # it is a unstake + transfer  # noqa: E501
+            event_type, event_subtype = HistoryEventType.RECEIVE, HistoryEventSubType.NONE
+            notes = 'Received {amount} {symbol} unstaked at EigenLayer' + f' from {depositor}'
+        elif depositor_is_tracked is True and withdrawer_is_tracked is False:  # we withdraw to a different address not tracked  # noqa: E501
+            notes = 'Sent {amount} {symbol} unstaked at EigenLayer' + f' to {withdrawer}'
+            event_type, event_subtype = HistoryEventType.SPEND, HistoryEventSubType.NONE
+        else:
+            log.error(f'Unexpected eigenlayer withdrawal in {context.transaction.tx_hash.hex()}. Skipping')  # noqa: E501
+            return DEFAULT_DECODING_OUTPUT
+
+        for event in context.decoded_events:
+            if event.event_type == HistoryEventType.RECEIVE:
+                event.event_type = event_type
+                event.event_subtype = event_subtype
+                event.counterparty = CPT_EIGENLAYER
+                event.product = EvmProduct.STAKING
+                event.notes = notes.format(amount=event.balance.amount, symbol=event.asset.resolve_to_crypto_asset().symbol)  # noqa: E501
+                break
```

Comment:
add a log error in the else for the for as this is an error

```suggestion
                break

        else:
            log.error(f'Could not match eigenlayer withdrawal event in {context.transaction.tx_hash.hex()}')
```